dor/providers/parsers.py

```diff
@@ -59,45 +65,51 @@ def get_data_files(self) -> list[FileMetadata]:
             for elem in self.tree.findall(".//METS:file")
         ]
 
-    def get_md_file_metadatum(self, elem):
+    def get_md_file_metadatum(self, elem: ElementAdapter):
         id_ = elem.get("ID")
         use = elem.get("USE")
-        locref = elem.find("METS:mdRef").get_optional("LOCREF")
-        mdtype = elem.find("METS:mdRef").get_optional("MDTYPE")
-        mimetype = elem.find("METS:mdRef").get_optional("MIMETYPE")
+        md_red_element = elem.find("METS:mdRef")
+        
+        if md_red_element:
+            locref = self.file_provider.apply_relative_path(self.data_path, md_red_element.get_optional("LOCREF"))
+            mdtype = md_red_element.get_optional("MDTYPE")
+            mimetype = md_red_element.get_optional("MIMETYPE", None)
 
         return FileMetadata(
             id=id_,
             use=use,
-            ref=FileReference(locref=locref, mdtype=mdtype, mimetype=mimetype),
+            ref=FileReference(locref=str(locref), mdtype=mdtype, mimetype=mimetype),
         )
 
-    def get_filesec_file_metadatum(self, elem):
+    def get_filesec_file_metadatum(self, elem: ElementAdapter):
         id_ = elem.get("ID")
         use = elem.get("USE")
         mdid = elem.get_optional("MDID", None)
         groupid = elem.get_optional("GROUPID", None)
-        mimetype = elem.get('MIMETYPE')
+        mimetype = elem.get_optional('MIMETYPE', None)
         mdtype = None
-        locref = elem.find("METS:FLocat").get_optional("LOCREF")
+        flocat_element = elem.find("METS:FLocat")
+        
+        if flocat_element:
```

Comment:
Same note about not needing to check if `flocat_element` is truthy, an exception will be raised if nothing is found.